api/error_injection.cc

```diff
@@ -24,10 +26,27 @@ void set_error_injection(http_context& ctx, routes& r) {
     hf::enable_injection.set(r, [](std::unique_ptr<request> req) {
         sstring injection = req->param["injection"];
         bool one_shot = req->get_query_param("one_shot") == "True";
-        auto& errinj = utils::get_local_injector();
-        return errinj.enable_on_all(injection, one_shot).then([] {
-            return make_ready_future<json::json_return_type>(json::json_void());
-        });
+        auto params = req->content;
+
+        if (params.size() > 1024 * 1024) {
+            // This is a hard limit, because we don't want to allocate
+            // too much memory or block the thread for too long.
+            throw httpd::bad_param_exception("Injection parameters too long");
```

Comment:
Injection parameter is too long or Injection parameters are too long

Reply:
Better reflect the limit in the message, ee.g. Injection parameters are too long, max length is X